### ISSUE TITLE
Add "capabilities" endpoint

### ISSUE THREAD

**Issue by vgeorge** [developmentseed/observe-api#23]
This endpoint would return the API version and required client versions. Example response:

```json
{
  "version": "0.1.0",
  "requiredClientVersion": {
     "mobile": ">0.10",
     "web": ">0.3"
  } 
}
```

Does this make sense? Should we this endpoint `/capabilities` or do you prefer a better name?

cc @developmentseed/observe

**Comment by geohacker** [comment]
@vgeorge /capabilities sounds good to me 👍

**Comment by batpad** [comment]
This looks good to me. My concern with only this approach is deciding when the client should make a call to the `capabilities` endpoint - we obviously don't want to make this call before every request.

Just thinking out loud:

What if the client were to send a header with every request - like `X-CLIENT-VERSION` with a value like `v1.1.0` or whatever the version is. Then, the backend has middleware or so that checks every request, and if the version mentioned in the header is below the supported client version, it returns an error.

It just seems to me to be slightly hard to decide when we should make the capabilities call from the client, and this would solve for that - and seems pretty easy to hook up for every request both on the client and server. Of course, the down-side is sending this extra header with every request, but that's definitely much better than having to call `/capabilities` before each request.

**Comment by vgeorge** [comment]
I agree that it's better to include a header with the client version. Is `X-CLIENT-VERSION` part of some standard? Does it make sense to return status [426 (Upgrade required)](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Status/426) when the client is outdated?

**Comment by batpad** [comment]
> Is X-CLIENT-VERSION part of some standard?

Nope, I just made that up. Will look around a bit to see if there's a standard way of doing this. Otherwise prefixing with `X-` and sending custom headers is _okay_, I think.

> Does it make sense to return status 426 (Upgrade required) when the client is outdated?

Hmm, the docs seem to indicate 426 is pretty specifically to request a protocol upgrade (to try and force the client to use HTTP 2.0 for eg) - it seems convenient, but I wonder if browsers may do weird things in response to a 426, if the purpose is specifically to request a protocol upgrade, which this is not?

A `400` with a specific error message seems okay to me.

In general though, it would be nice to do some research and see if there's a common pattern to solve this problem / if there are some standard-ish ways of doing this. I can do some looking up tomorrow.
